### PR TITLE
Update to dotnet 6

### PR DESCRIPTION
**Pull request by rido-min**
Update TargetFramework, and packages dependencies
Update DockerFiles

### CODE REVIEW

**Comment by marianan** [comment]
@rido-min thanks so much for your PR. Could you also test the changes using this quickstart https://github.com/Azure/iotedgedev/blob/main/docs/quickstart.md?

**Comment by rido-min** [comment]
@marianan iotedgedev uses the published dotnet template, I guess I have to wait until the package is available in NuGet.

In the meantime I can test with the current IoTEdge VS Extension by adding a custom project.

**Comment by marianan** [comment]
My mistake, the instructions we should be validating with are here under the CLI section https://learn.microsoft.com/en-us/azure/iot-edge/how-to-vs-code-develop-module?view=iotedge-1.4&tabs=csharp&pivots=iotedge-dev-cli . You are right, that instead of itoedgedev init, we should use: 'dotnet new -i  path-to-local-nupkg' 'dotnet new aziotedgemodule -n <module_name>'

**Comment by marianan** [comment]
Please make sure to increment the version at https://github.com/Azure/dotnet-template-azure-iot-edge-module/blob/0105f01759c0cfe997a2be2974a195d91407ffe9/Microsoft.Azure.IoT.Edge.Module.nuspec#L5 , https://github.com/Azure/dotnet-template-azure-iot-edge-module/blob/0105f01759c0cfe997a2be2974a195d91407ffe9/CHANGELOG.md?plain=1#L5

**Comment by marianan** [comment]
Does anything need to change in the test setup csproj?  https://github.com/Azure/dotnet-template-azure-iot-edge-module/blob/0105f01759c0cfe997a2be2974a195d91407ffe9/Test/Test.csproj#L1 ?

**Comment by rido-min** [comment]
version updated to 3.2.0, and Tests are now targeting 6.0

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 [![Nuget](https://img.shields.io/nuget/v/Microsoft.Azure.IoT.Edge.Module.svg)](https://www.nuget.org/packages/Microsoft.Azure.IoT.Edge.Module/)
 
+### 3.2.0 (2022-11-29)
+* [Updated] Update template to target dotnet6
```
Comment (marianan) [inline CHANGELOG.md:6]: Please mention the worker changes too

**Review by marianan — APPROVED** [review]

File: content/dotnet-template-azure-iot-edge-module/FSharp/Program.fs [hunk 2]
```diff
@@ -20,8 +20,8 @@ module SampleModule =
     let PipeMessage (message:Message) (userContext:obj) =
         let counterValue = Interlocked.Increment(counter)
 
-        let deviceClient = userContext :?> DeviceClient
```
[the remark this answers is not in the document]
Comment (rido-min) [reply content/dotnet-template-azure-iot-edge-module/FSharp/Program.fs:23]: The C# changes have been reviewed by DrWill. 
I'm not updating F# template to be based on the Worker pattern.

**Pull request merged**